sensors/src/test/scala/com/lynbrookrobotics/potassium/sensors/imu/TestXYPosition.scala

```diff
@@ -0,0 +1,88 @@
+package com.lynbrookrobotics.potassium.sensors.imu
+
+import com.lynbrookrobotics.potassium.Signal
+import com.lynbrookrobotics.potassium.sensors.position.xyPosition
+import com.lynbrookrobotics.potassium.units.Point
+import org.scalatest.FunSuite
+import squants.time.Milliseconds
+import squants.motion.{DegreesPerSecond, FeetPerSecond}
+import squants.space.{Degrees, Feet}
+import squants.time.Seconds
+
+class TestXYPosition extends FunSuite{
+  val origin = new Point(Feet(0), Feet(0))
+
+  test("Test moving straight at 2 ft/s for 1 sec results in moving 2 ft forward") {
+    lazy val startTime = System.currentTimeMillis
+    def timePassed = Milliseconds(System.currentTimeMillis - startTime)
+
+    val distance = Signal(timePassed * FeetPerSecond(2))
+    val targetPosition = new Point(Feet(0), Feet(2))
+
+    val position = xyPosition(
+      origin,
+      Signal.constant(Degrees(90)),
+      distance.toPeriodic
+    )
+
+    var i = 0
+    while (timePassed <= Seconds(1)){
+      if(i % 10000000 == 0) position.currentValue(Seconds(0.1))
```

Comment:
Yeah, not really sure what you're trying to do here. You should just be able to run `position.currentValue(Seconds(0.1))` 10 times to achieve a simulated duration of 1 second.